src/components/AccountModule/useClientBlockNumber.js

```diff
@@ -0,0 +1,33 @@
+import React, { useState, useEffect, useContext } from 'react'
+
+import { web3Providers } from '../../environment'
+import { pollEvery } from '../../utils'
+import { getWeb3 } from '../../web3-utils'
+
+const CLIENT_WEB3 = getWeb3(web3Providers.default)
+const CLIENT_BLOCK_POLL_DELAY = 5000
+
+const ClientBlockNumberContext = React.createContext()
+
+export function ClientBlockNumberProvider(props) {
```

Comment:
Hmm, does it make sense to pass the props through to the `Provider`?

Reply:
I wrote it that way due to `wallet.js` `WalletBlockNumberProvider` also having the `props` arg. 🙂